addons/stock_account/models/product.py

```diff
@@ -176,13 +182,75 @@ def _sum_remaining_values(self):
         return sum(moves.mapped('remaining_value'))
 
     @api.multi
-    @api.depends('stock_move_ids.product_qty', 'stock_move_ids.state', 'product_tmpl_id.cost_method')
+    @api.depends('stock_move_ids.product_qty', 'stock_move_ids.state', 'stock_move_ids.remaining_value', 'product_tmpl_id.cost_method', 'product_tmpl_id.standard_price', 'product_tmpl_id.property_valuation', 'product_tmpl_id.categ_id.property_valuation')
     def _compute_stock_value(self):
+        StockMove = self.env['stock.move']
+        to_date = self.env.context.get('to_date')
+
+        fifo_automated_values = {}
+        if to_date:
+            query = """SELECT aml.product_id, aml.account_id, sum(aml.debit) - sum(aml.credit), sum(quantity), array_agg(aml.id)
+                         FROM account_move_line AS aml
+                   INNER JOIN account_move AS am ON (aml.move_id = am.id)
+                        WHERE am.stock_move_id IS NOT NULL AND am.date <= %s
+                     GROUP BY aml.product_id, aml.account_id"""
+            params = (to_date,)
```

Comment:
in the WHERE clause filter on aml account_id current and previous valuation's account
tricky in sql (join with categ_id and ir.property and play around the res_value column) so a previous sql/orm query will do the trick

without this change if the user updates its valuation account the value aren't correct